syncmap_test.go

```diff
@@ -0,0 +1,212 @@
+		"    int(1): int(100)",
+		"    int(2): int(200)",
+		"}",
+	)
+}
+
+// This test verifies the formatting of sync.Map key/values in the named structs.
+func TestPrinter_SyncMapInNamedStruct(t *testing.T) {
+	test(
+		t,
+		"empty sync.Map",
+		&syncmaps{},
+		"*github.com/dogmatiq/dapper_test.syncmaps{<zero>}",
+	)
+
+	sm := &syncmaps{}
+
+	sm.Map.Store(1, 100)
+	sm.Map.Store(2, 200)
+
+	test(
+		t,
+		"non-empty sync.Map",
+		sm,
+		"*github.com/dogmatiq/dapper_test.syncmaps{",
+		"    Map: {",
+		"        int(1): int(100)",
+		"        int(2): int(200)",
+		"    }",
+		"}",
+	)
+}
+
+// This test verifies that sync.Map keys are sorted by their formatted string
+// representation.
+func TestPrinter_SyncMapKeySorting(t *testing.T) {
+	var m sync.Map
+
+	m.Store("foo", 1)
+	m.Store("bar", 2)
+
+	test(
+		t,
+		"keys are sorted by their string representation",
+		&m,
+		"*sync.Map{",
+		`    "bar": int(2)`,
+		`    "foo": int(1)`,
+		"}",
+	)
+}
+
+// This test verifies that values associated with sync.Map keys that have a
+// multiline string representation are aligned correctly.
+func TestPrinter_MultilineSyncMapKeyAlignment(t *testing.T) {
+	var m sync.Map
+
+	m.Store("short", "one")
+	m.Store("the longest key in the galaxy must be longer than it was before", "two")
+	m.Store(multiline{Key: "multiline key"}, "three")
+
+	test(
+		t,
+		"keys are aligned correctly",
+		&m,
+		"*sync.Map{",
+		`    "short":                                                           "one"`,
+		`    "the longest key in the galaxy must be longer than it was before": "two"`,
+		"    github.com/dogmatiq/dapper_test.multiline{",
+		`        Key: "multiline key"`,
+		`    }: "three"`,
+		"}",
+	)
+
+	m.Delete("the longest key in the galaxy must be longer than it was before")
+
+	test(
+		t,
+		"keys are aligned correctly when the longest line is part of a multiline key",
+		&m,
+		"*sync.Map{",
+		`    "short":                                   "one"`,
+		"    github.com/dogmatiq/dapper_test.multiline{",
+		`        Key: "multiline key"`,
+		`    }: "three"`,
+		"}",
+	)
+}
+
+// This test verifies that recursive sync.Map is detected, and do not produce
+// an infinite loop or stack overflow.
+func TestPrinter_SyncMapRecursion(t *testing.T) {
+	var m sync.Map
+	m.Store("child", &m)
+
+	test(
+		t,
+		"recursive sync.Map",
+		&m,
+		"*sync.Map{",
+		`    "child": *sync.Map(<recursion>)`,
+		"}",
+	)
+}
+
+// testFilterPrinter is the test implementation of FilterPrinter interface.
+type testFilterPrinter struct {
+	WriteFn          func(io.Writer, Value) error
+	FormatTypeNameFn func(Value) string
+}
+
+func (t *testFilterPrinter) Write(w io.Writer, v Value) error {
+	if t.WriteFn == nil {
+		return nil
+	}
+
+	return t.WriteFn(w, v)
+}
+
+func (t *testFilterPrinter) FormatTypeName(v Value) string {
+	if t.FormatTypeNameFn == nil {
+		return ""
+	}
+
+	return t.FormatTypeNameFn(v)
+}
+
+// This test verifies that recursive sync.Map is detected, and do not produce
```

Comment:
Is this the right doc block for this test function? It seems to be just a copy of the one above and not related to this test.

Reply:
That's true, thanks man. After #37 is merged, I can delete this entire test including this mismatched comment altogether.